Realm/RLMSyncSession.h

```diff
@@ -150,8 +152,20 @@ NS_ASSUME_NONNULL_BEGIN
  */
 - (nullable RLMProgressNotificationToken *)addProgressNotificationForDirection:(RLMSyncProgressDirection)direction
                                                                           mode:(RLMSyncProgress)mode
+                                                           dispatchToMainQueue:(BOOL)dispatchToMainQueue
```

Comment:
If we're okay with telling users they must explicitly dispatch onto whatever queue they want, sure. (We might also not want users to 'accidentally' run expensive work in their callbacks without dispatching, since IIRC the callback is run on the sync client's thread.)